soundata/datasets/tau2019uas.py

```diff
@@ -0,0 +1,661 @@
+
+    @property
+    def split(self):
+        return self._clip_metadata.get("split")
+
+    @property
+    def tags(self):
+        scene_label = self._clip_metadata.get("scene_label")
+        if scene_label is None:
+            return None
+        else:
+            return annotations.Tags([scene_label], np.array([1.0]))
+
+    @property
+    def city(self):
+        return self._clip_metadata.get("city")
+
+    @property
+    def identifier(self):
+        return self._clip_metadata.get("identifier")
+
+    def to_jams(self):
+        """Get the clip's data in jams format
+
+        Returns:
+            jams.JAMS: the clip's data in jams format
+
+        """
+        return jams_utils.jams_converter(
+            audio_path=self.audio_path, tags=self.tags, metadata=self._clip_metadata
+        )
+
+
+@io.coerce_to_bytes_io
+def load_audio(fhandle: BinaryIO, sr=None) -> Tuple[np.ndarray, float]:
+    """Load a  TAU Urban Acoustic Scenes 2019 audio file.
+
+    Args:
+        fhandle (str or file-like): File-like object or path to audio file
+        sr (int or None): sample rate for loaded audio, None by default, which
+            uses the file's original sample rate of 44100 without resampling.
+
+    Returns:
+        * np.ndarray - the mono audio signal
+        * float - The sample rate of the audio file
+
+    """
+    audio, sr = librosa.load(fhandle, sr=sr, mono=False)
+    return audio, sr
+
+
+@core.docstring_inherit(core.Dataset)
+class Dataset(core.Dataset):
+    """
+    The  TAU Urban Acoustic Scenes 2019 dataset
+    """
+
+    def __init__(self, data_home=None):
+        super().__init__(
+            data_home,
+            name="tau2019uas",
+            clip_class=Clip,
+            bibtex=BIBTEX,
+            remotes=REMOTES,
+            license_info=LICENSE_INFO,
+        )
+
+    @core.copy_docs(load_audio)
+    def load_audio(self, *args, **kwargs):
+        return load_audio(*args, **kwargs)
+
+    @core.cached_property
+    def _metadata(self):
+
+        metadata_path = os.path.join(
+            self.data_home, "TAU-urban-acoustic-scenes-2019-development", "meta.csv"
+        )
+
+        splits = [
+            "development.train",
+            "development.evaluate",
+            "development.test",
+            "evaluation",
+            "leaderboard",
+        ]
+
+        metadata_index = {}
+
+        with open(metadata_path) as csv_file:
+            csv_reader = csv.reader(csv_file, delimiter="\t")
+            next(csv_reader)
+            for row in csv_reader:
+                file_name = os.path.basename(row[0])
+                clip_id = os.path.basename(file_name).replace(".wav", "")
+                scene_label = row[1]
+                identifier = row[2]
+                city = identifier.split("-")[0]
+                metadata_index[clip_id] = {
+                    "scene_label": scene_label,
+                    "city": city,
+                    "identifier": identifier,
```

Comment:
I'm a bit confused about the repetition of code in `_metadata`, can you merge (if there are differences) this block with the one below?

Reply:
The problem here is that some information is in the development metadata (e.g the identifier), some is in the evaluation setup files (e.g. split), and some is in both places (e.g. city, scene_label). I've decided to use metadata file for everything (first round) and update the split looking at the evaluation setup files (second round). Besides, in the second round, I have to define the metadata_index for evaluation and leaderboard datasets (they don't have metadata file). @magdalenafuentes  do you find a better way to do it?

Reply:
mmmm... ok I see. I was also a bit worried that you might be overwriting the metadata_index dictionary if you have repeated `clip_id`, but I guess this would be completely avoided with your new naming convention. I'm good leaving this like it is!

Reply:
Yes, actually I was overwriting it, but as you said, now should be fixed.